### PR TITLE
New example: Prometheus sink

### PR DESCRIPTION
**Pull request by nicusX**
## Purpose of the change

New example: Prometheus sink connector

## Verifying this change

* Tested on Managed Flink + AMP

## Significant changes

*(Please check any boxes [x] if the answer is "yes". You can first publish the PR and check them afterward, for convenience.)*

- [X] Completely new example
- [ ] Updated an existing example to newer Flink version or dependencies versions
- [ ] Improved an existing example
- [ ] Modified the runtime configuration of an existing example (i.e. added/removed/modified any runtime properties)
- [ ] Modified the expected input or output of an existing example (e.g. modified the source or sink, modified the record schema)

### CODE REVIEW

File: java/PrometheusSink/README.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+Runtime parameters:
+
+| Group ID        | Key                 | Description                                                                                                                                   | 
+|-----------------|---------------------|-----------------------------------------------------------------------------------------------------------------------------------------------|
+| `DataGen`       | `records.per.sec`   | (optional) Number of generated records per second. Default = 100.                                                                             
+| `PrometheusSink` | `endpoint.url`      | URL of the Remote-Write endpoint of the Prometheus workspace                                                                                  |
+| `PrometheusSink` | `aws.region`        | Region of AMP workspace. If not specified, it will use the application region or the default region of the AWS profile, when running locally. |
+| `PrometheusSink` | `max.request.retry` | Maximum retryes in case AMP respond with a retriable error (e.g. throttling).                                                                 |
```
Comment (hlteoh37) [inline java/PrometheusSink/README.md:47]: nit: `retries`

File: java/PrometheusSink/pom.xml [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+            <artifactId>flink-clients</artifactId>
+            <version>${flink.version}</version>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.flink</groupId>
+            <artifactId>flink-runtime-web</artifactId>
+            <version>${flink.version}</version>
+            <scope>provided</scope>
+        </dependency>
```
Comment (hlteoh37) [inline java/PrometheusSink/pom.xml:45]: Do we know why we need this?

File: java/PrometheusSink/pom.xml [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+
+    <properties>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <jar.finalName>${project.name}-${project.version}</jar.finalName>
+        <target.java.version>11</target.java.version>
+        <maven.compiler.source>${target.java.version}</maven.compiler.source>
+        <maven.compiler.target>${target.java.version}</maven.compiler.target>
+        <flink.version>1.20.0</flink.version>
+        <prometheus.connector.version>1.0.0-1.20</prometheus.connector.version>
+        <kda.runtime.version>1.2.0</kda.runtime.version>
```
Comment (hlteoh37) [inline java/PrometheusSink/pom.xml:20]: nit: can we use `msf`?

**Review by hlteoh37 — APPROVED** [review]
Added some nits! LGTM

File: java/PrometheusSink/pom.xml (hunk 3 above)
Comment (nicusX) [reply java/PrometheusSink/pom.xml:20]: No, until we change the name of the artifact that is called "KinesisAnalytics runtime"

File: java/PrometheusSink/pom.xml (hunk 2 above)
Comment (nicusX) [reply java/PrometheusSink/pom.xml:45]: For having the Flink UI locally. It's handy

**Pull request merged**